src/apollo/cyber/proto/CMakeLists.txt

```diff
@@ -19,10 +19,11 @@ set(PROTOS choreography_conf.proto
 
 PROTOBUF_GENERATE_CPP(PROTO_SRCS PROTO_HDRS ${PROTOS})
 
-include_directories(${PROTOBUF_INCLUDE_DIRS})
 
 add_library(cyber_proto ${PROTO_SRCS} ${PROTO_HDRS})
 target_link_libraries(cyber_proto ${PROTOBUF_LIBRARIES})
+target_include_directories(cyber_proto PUBLIC ${PROTOBUF_INCLUDE_DIRS}
+  INTERFACE ${CMAKE_INSTALL_PREFIX})
```

Comment:
why `INTERFACE` ? 
Use generator Expressions for this?
```
$<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}>
```

Reply:
Key word `INTERFAE` specifies the installation path, So I think `INTERFACE` is equal to `$<INSTALL_INTERFACE:xxx>`. When the key word is `PUBLIC`, generator expressions are often used to specify building paths and installation paths respectively. Such as `$<BUILDING_INTERFACE:xxx>` and `$<INSTALL_INTERFACE:xxx>`

Reply:
Yes. They are equal in terms of effects. The reason you add INTERFACE is that you want users to find these header files in config mode, right? This is clearly a usage requirement. These headers are also public. So using  `$<INSTALL_INTERFACE:xxx>` is more clear.